catroid/src/main/java/org/catrobat/catroid/ui/WebViewActivity.java

```diff
@@ -189,8 +168,21 @@ public boolean onKeyDown(int keyCode, KeyEvent event) {
 	}
 
 	private class MyWebViewClient extends WebViewClient {
+
 		@Override
 		public void onPageStarted(WebView view, String urlClient, Bitmap favicon) {
+			if (progressCircle == null) {
+				progressCircle = new ProgressDialog(view.getContext(), R.style.WebViewLoadingCircle);
+				progressCircle.setCancelable(true);
+				progressCircle.setCanceledOnTouchOutside(false);
+				progressCircle.setProgressStyle(android.R.style.Widget_ProgressBar_Small);
+				try {
+					progressCircle.show();
+				} catch (Exception e) {
```

Comment:
![challenge_accepted](https://cloud.githubusercontent.com/assets/13980635/24218183/5c93d6d8-0f68-11e7-8357-47ade8150c8f.jpg)

Will do! I'm done with all changes, so i'll make a commit for the same, then clean my commits! Sorry for the trouble!